autoscale-docker-swarm-container/README.md

```diff
@@ -0,0 +1,167 @@
+    Default: `300`  
+    Used to specify the amount of time, in seconds, that we ideally want to complete the current backlog of messages in. This is effectively a quality of service parameter, where lower will trigger more aggressive scaling.
+
+ - `CAF_AUTOSCALER_MAXIMUM_INSTANCES`  
+    Default: `100`  
+    Used to specify the maximum number of instances that any worker can be scaled to.
+
+ - `CAF_DOCKER_SWARM_TIMEOUT`  
+    Default: `30`  
+    Used to specify the max length of time in seconds that a docker REST call can take before a timeout out occurs.  
+
+ -  `CAF_DOCKER_SWARM_HEALTHCHECK_TIMEOUT`  
+    Default: `5`  
+    Used to specify the max length of time in seconds that the Docker endpoint healthcheck can take before a timeout occurs.
+
+ - `HTTP_PROXY`  
+    (Optional)
+    Used to specify an HTTP based proxy, for the Docker REST endpoint communication. 
+
+### Health checks
+
+The [autoscale-core](https://github.com/Autoscaler/autoscaler/tree/develop/autoscale-core) application inherently exposes standard and
+module-specific health checks. If you expose the admin port (default 8081)
+then this can be accessed via HTTP to examine metrics and health checks.
+The health check REST call will return HTTP 500 if any health check fails.
+For details on the health checks for specific components, examine the module
+documentation.
+
+
+### Logging
+
+Currently the logging is accessible via the Marathon sandbox until a CAF
+approved logging mechanism is agreed upon. Debug logging is enabled by default
+in this container.
+
+
+### Scaling
+
+This container is not election-enabled. This means there should only be one
+instance per set of services the container is scaling. If the container is
+being run on Docker Swarm, this will ensure the container is restarted if it
+crashes. It is recommended you do not attempt to monitor and scale a very
+large number of applications with a single instance. While this should work
+with an arbitrary number of target services, the monitoring and scaling
+actions may begin to lag behind the requested period once the container goes
+beyond its normal capability.
+
+
+### Resource requirements
+
+The container uses very little memory and CPU, and effectively no disk load.
+It does, however, generate some network traffic through monitoring and scaling
+of services. Docker Swarm will be the target of most of the network traffic, though
+the RabbitMQ management host will also be polled. The frequency of the traffic
+depends upon the number of services being monitored and the service template
+requested frequency. Each individual network call should be no more than a
+handful of kilobytes, though the 15-minute scheduled refresh of all Docker swarm
+services may be more if Docker Swarm is running a large number of services.
+Recommended values:
+
+- 128-196MB of RAM
+- 0.1 CPUs per 10 services monitored
+- Slow I/O is acceptable
+- Preferred to be network-local to Marathon and RabbitMQ
```

Comment:
Agreed, but the resource requirements is useful.  I see that production type compose files are being produced which correctly start to give these types of values for resources in the templates.
Should I be putting the same memory requirements as used in marathons version into the default docker-compose file example in the swarm container?  https://github.com/Autoscaler/autoscaler/blob/CAF-2191/autoscale-docker-swarm-container/src/main/deployment/docker-compose.yml